Run kazel on the entire tree

ixdy opened the pull request:
**What this PR does / why we need it**: part of #47558: auto-generate `BUILD` files on the entire tree, since this is what `gazelle` does, and it'll make subsequent reviews easier if less is changing.

**Release note**:
<!--  Steps to write your release note:
1. Use the release-note-* labels to set the release note state (if you have access)
2. Enter your extended release note in the below block; leaving it blank means using the PR title as the release note. If no release note is required, just write `NONE`.
-->
```release-note
NONE
```
/assign
/release-note-none

ixdy committed:
Update kazel to include kubernetes/repo-infra#21

ixdy committed:
Update kazelcfg to kazel everything

ixdy committed:
Fix BUILD files

ixdy committed:
Make hack/boilerplate/test files use a more appropriate package name

ixdy commented:
/unassign
/assign @mikedanese @spxtr 

Seems to work.

ixdy commented:
(you might want to review individual commits and ignore the autogenerated one)

ixdy commented:
This bazel failure is weird:
```
W0719 23:22:33.712] ERROR: /workspace/k8s.io/kubernetes/staging/src/k8s.io/apimachinery/pkg/util/sets/BUILD:25:1: output 'staging/src/k8s.io/apimachinery/pkg/util/sets/byte.go' was not created.
W0719 23:22:33.716] ERROR: /workspace/k8s.io/kubernetes/staging/src/k8s.io/apimachinery/pkg/util/sets/BUILD:25:1: output 'staging/src/k8s.io/apimachinery/pkg/util/sets/doc.go' was not created.
W0719 23:22:33.718] ERROR: /workspace/k8s.io/kubernetes/staging/src/k8s.io/apimachinery/pkg/util/sets/BUILD:25:1: output 'staging/src/k8s.io/apimachinery/pkg/util/sets/empty.go' was not created.
W0719 23:22:33.720] ERROR: /workspace/k8s.io/kubernetes/staging/src/k8s.io/apimachinery/pkg/util/sets/BUILD:25:1: output 'staging/src/k8s.io/apimachinery/pkg/util/sets/int.go' was not created.
W0719 23:22:33.721] ERROR: /workspace/k8s.io/kubernetes/staging/src/k8s.io/apimachinery/pkg/util/sets/BUILD:25:1: output 'staging/src/k8s.io/apimachinery/pkg/util/sets/int64.go' was not created.
W0719 23:22:33.723] ERROR: /workspace/k8s.io/kubernetes/staging/src/k8s.io/apimachinery/pkg/util/sets/BUILD:25:1: output 'staging/src/k8s.io/apimachinery/pkg/util/sets/string.go' was not created.
W0719 23:22:33.730] ERROR: /workspace/k8s.io/kubernetes/staging/src/k8s.io/apimachinery/pkg/util/sets/BUILD:25:1: not all outputs were created or valid.
```

It's already the case in upstream/master that doing `bazel buildstaging/src/k8s.io/apimachinery/pkg/util/sets/...` fails in the same way, whereas `bazel build //vendor/k8s.io/apimachinery/pkg/util/sets/...` works as expected.
I'm not sure what changed here to break this.

ixdy commented:
I have independently discovered that @jmillikin-stripe's #48565 fixes the bazel failure.

ixdy commented:
/rtests
Tests are so flaky. At least bazel test is working now.

ixdy commented:
/retest

this time with correct spelling

ixdy commented on hack/BUILD:
```diff
@@ -47,3 +48,22 @@ test_suite(
 )
+
+go_binary(
+    name = "hack",
```
this one is probably better named `e2e` but I'm guessing gazel wouldn't support that.

ixdy commented on staging/BUILD:
```diff
@@ -0,0 +1,214 @@
+)
+
+go_binary(
+    name = "staging",
```
this one is also mis-named.

ixdy commented:
other than the two `go_binary`s I mentioned, the autogenerated stuff looks as expected.

spxtr commented:
Is this ready to go in?

ixdy commented:
yes, I'm waiting for review and then I'll hunt down a TLA.

ixdy commented:
(I just rebased + regenerated. hopefully it all still works. :)

mikedanese commented:
/lgtm

ixdy commented:
/assign @thockin 

can I get TLA for the BUILD file changes?

ixdy commented:
/retest

thockin commented:
/approve

ixdy commented:
@k8s-merge-robot wake up, kops is green

ixdy commented:
/test pull-kubernetes-e2e-kops-aws

ixdy commented:
/retest

ixdy commented:
/retest

On Aug 1, 2017 20:10, "k8s-ci-robot" <notifications@github.com> wrote:

> @ixdy <https://github.com/ixdy>: The following test *failed*, say /retest
> to rerun them all:
> Test name Commit Details Rerun command
> pull-kubernetes-e2e-kops-aws c0405ea
> <https://github.com/kubernetes/kubernetes/commit/c0405ea672f8641073420a39478739bf5cc73523>
> link
> <https://k8s-gubernator.appspot.com/build/kubernetes-jenkins/pr-logs/pull/48969/pull-kubernetes-e2e-kops-aws/38792/> /test
> pull-kubernetes-e2e-kops-aws
>
> Full PR test history <https://k8s-gubernator.appspot.com/pr/48969>. Your
> PR dashboard <https://k8s-gubernator.appspot.com/pr/ixdy>. Please help us
> cut down on flakes by linking to
> <https://github.com/kubernetes/community/blob/master/contributors/devel/flaky-tests.md#filing-issues-for-flaky-tests>
> an open issue
> <https://github.com/kubernetes/kubernetes/issues?q=is:issue+is:open> when
> you hit one in your PR.
>
> Instructions for interacting with me using PR comments are available here
> <https://github.com/kubernetes/community/blob/master/contributors/devel/pull-requests.md>.
> If you have questions or suggestions related to my behavior, please file an
> issue against the kubernetes/test-infra
> <https://github.com/kubernetes/test-infra/issues/new?title=Prow%20issue:>
> repository. I understand the commands that are listed here
> <https://github.com/kubernetes/test-infra/blob/master/commands.md>.
>
> —
> You are receiving this because you were mentioned.
> Reply to this email directly, view it on GitHub
> <https://github.com/kubernetes/kubernetes/pull/48969#issuecomment-319555412>,
> or mute the thread
> <https://github.com/notifications/unsubscribe-auth/AHsBcPY5IVHde1SW7FH3SVwaqkD38RC9ks5sT-iygaJpZM4OY05K>
> .
>

ixdy commented:
/retest

ixdy committed:
Run hack/update-bazel.sh to generate BUILD files

ixdy commented:
Rebased and regenerated.

ixdy commented:
/retest

spxtr commented:
[Are we really that flaky right now?](https://k8s-gubernator.appspot.com/pr/48969) Jeez.

ixdy commented:
🎄🎄🎄

ixdy commented:
(still need a re-LGTM)

spxtr commented:
/lgtm

ixdy commented:
Rebased and regenerated *again*. ARGH.

ixdy commented:
only change since last lgtm tag was regenerating the BUILD files. I've checked that commit manually and it looks as expected, so retagging so this can merge overnight before it breaks again.

ixdy commented:
/retest
